### ISSUE TITLE
New Microsoft launcher stops authenticating and passing through credentials to game to start

### ISSUE THREAD

**Issue by the1xaks** [FTBTeam/FTB-App#491]
This had been working fine since the initial "migrate to the Microsofy Launcher" (that nobody wants). I'm currently playing Direwolf20s 1.16 pack, and I'm ~400ish hours in. I have a hosted server on Bisect that I have my world on. This was just fine last night.

This morning, when I try to log in, I get a prompt that there's a problem, and I need to re-authenticate with Satya (pics 1 & 2)

So I follow the prompts and get told I'm good to go. My creds are fine and should be good to go (pic 3)

Only the next thing I see is a nag screen (pic 4) that says "Satya thanks you for your wallet, sucka"

I noted there's some other similar bugs listed, so I checked a few things like: wiping cookies out and reloading, for fresh cookie creation.  I checked to make sure MS has my actual personal data, so it won't think I'm under 18 for some reason. My default language hasn't changed, nor have any updates run on my local PC account.

**Comment by the1xaks** [comment]
![MC_pic2](https://user-images.githubusercontent.com/88561682/157472270-7592c11f-0965-49b6-8dad-d430d572849b.png)
![MC_pic3](https://user-images.githubusercontent.com/88561682/157472271-d14d7ed0-144e-48b7-9de9-6b077ced8385.png)
![MC_pic4](https://user-images.githubusercontent.com/88561682/157472272-89c53d7e-f680-45f7-a267-6c17ac3300a2.png)
![MC_pic1](https://user-images.githubusercontent.com/88561682/157472275-98df6c4f-06a4-4fdf-8ff6-37a7f4f97293.png)

**Comment by the1xaks** [comment]
Did snap a screencap of a quick little launcher popup during the process. It goes away really fast

![MC_pic5](https://user-images.githubusercontent.com/88561682/157473260-36197402-5c4a-4340-873d-a0f2d10b944b.png)

**Comment by the1xaks** [comment]
Odd detail

If you use another launcher wrapper process, say, the CurseForge Overwolf app - it works as normal. I'm loading into All The Mods 7 as I type this. It authenticates to the Microsoft servers correctly no problem

**Comment by MichaelHillcox** [comment]
Are you using an under 18s microsoft account?

**Comment by the1xaks** [comment]
I specifically stated that I checked the MS side of the account to make
sure they had my correct birthday and age information in the thread.

Yes, they have my correct age attached to the MS account. I'm 47, which is
significantly over 18

**Comment by MichaelHillcox** [comment]
> Only the next thing I see is a nag screen (pic 4) that says "Satya thanks you for your wallet, sucka"

What?

Also, if you had followed the bug template you would have seen a link to our debug tool. Please see the options below and run them so I can see your issue. Please be sure to run them after you have attempted to login.

> This had been working fine since the initial "migrate to the Microsofy Launcher" (that nobody wants)

Not sure if you're pointing this at us then this was a required change. We've stopped using the Minecraft Launcher completely due to the backwards error messages, instability and constant problems. We in-turn now have to also handle a lot of the things the MC launcher did before on our own. Some things are a bit buggy and we are actively working on resolving this. 

If it's targeted at MC then yes, we agree, it's annoying.

**FTB Debug**
[Windows](https://dist.creeper.host/tools/ftb-debug/ftb-debug.exe) [MacOS](https://dist.creeper.host/tools/ftb-debug/ftb-debug-macos) [Linux](https://dist.creeper.host/tools/ftb-debug/ftb-debug-linux)

**Comment by the1xaks** [comment]
I do apologize, as Microsoft being...well, Microsoft, has been a burr under
my saddle for all of my 26 years in IT. Having them buy their way into my
gaming hobby just pisses me off, and I get prickly about it. That's my bad.
I'll try to not be so sophomoric going forward.

Bug template? I did not see that. I'm not a frequent user of the git site,
so I'll have to dig a bit deeper and see what else I missed. Lemme run that
real quick and get it in here for you:

[image: image.png]

The tool says it uploaded the logs...did you get what you needed, I hope?

---

I am intrigued by the comment you made, "Not sure if you're pointing this
at us then this was a required change. We've stopped using the Minecraft
Launcher completely due to the backwards error messages, instability and
constant problems."

I would LOVE to not use the MS launcher or account information AT ALL. Is
there a way I'm not aware of to bypass that terrible organization all
together and just use my Mojang creds (as we have collectively done for
years now) for running third-party modpacks? I'm playing All the Mods 7 via
the Curseforge Overwolf launcher as I type this...I zipped my FTB
Direwolf20 world into a backed up Instance on my server until this gets
squared up and the FTB launcher returns to normal functionality.

Please let me know if I've missed anything else you need to troubleshoot.
Also, again, please accept my apologies for being overly snarky earlier.
That's on me for being an ass today, not on you.

**Comment by MichaelHillcox** [comment]
Your image hasn't worked for some reason. I'm assuming it's because you've replied an email?

**Comment by the1xaks** [comment]
I pasted it inline

Lemme add it as an attachment, mayhap?

**Comment by the1xaks** [comment]
Let's try it THIS way, then?

**Comment by the1xaks** [comment]
![ftb_debug](https://user-images.githubusercontent.com/88561682/157998743-db548940-90a4-43fd-93f2-03586aea3b17.png)

**Comment by the1xaks** [comment]
Figured it out.

I have purchased OS (Windows 10) and purchased Office suite (2019) on this rig. I have those two items registered with Microsoft under email address me1@gmail.com with one specific set of credentials.

My Mojang account is tied to a separate MS account, for things that MS has bought and fucked up over the years that have ended up needing an MS identifier. This account, me2@gmail.com, exists at MS going all the way back to the Hotmail.com days, and came to be merged when I swapped from the old hotmail.com address to gmail as my main.

Last week, for some reason, the Microsoft launcher started trying to use my m1@gmail.com information and password. It was attempting to auto-login with these incorrect credentials. The MS account itself is fine, but it does not have my Minecraft info associated with it, just everything else.

I ended up uninstalling the FTB launcher from my PC, deleting the folders of that game instance, scrubbing the registry, rebooting, reinstalling, and then step by step re-registering the FTB launcher to the MS login process and verifying that it was using the me2@gmail.com credentials, instead of the me1@gmail.com from my Office and OS installs.

This restored the FTB launcher to working order.
